plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotBrokerPageSource.java

```diff
@@ -113,6 +121,9 @@ public Page getNextPage()
         int rowCount = 0;
         while (size < PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES && resultIterator.hasNext()) {
             rowCount++;
+            if (currentRowCount.incrementAndGet() > limitForBrokerQueries) {
```

Comment:
I understand `limitForBrokerQueries` is a temporary workaround until fix for https://github.com/apache/incubator-pinot/issues/7110 is widespread. Let's add a TODO comment here so that we can remove this.

Anyway, i don't quite understand why we want to do this.
We apply limit after the fact. We already know Pinot didn't crash and handled our query correctly.
Please store the explanation as a code comment.

Reply:
The limit could be applied manually in the passthrough query also, I'll add a comment and TODO.

Reply:
The other time a limit would be needed is during pushdown: no limit will only return 10 rows from pinot by default. If the limit is Integer.MAX_VALUE (limit is an integer in pinot) this can cause oom's in pinot (pinot#7110) -so it is set to the limit for broker queries config, and if there are more results an error is thrown.

Reply:
I'm a bit confused? Does Pinot stream results back as soon as they are available? i.e. does Pinot run into an OOM after some results have been returned? If not and Pinot OOMs early then what I think Piotr is saying is that this may be dead code since the Pinot query would've failed already and we'd not have reached here in the first place.

Can you share what Pinot's behaviour is?

Reply:
@elonazoulay Bump.